pyannote/database/custom.py

```diff
@@ -169,18 +169,21 @@ def subset_iter(database_name, file_lst=None, file_rttm=None,
         # initialize current file with the mandatory keys
         current_file = {'database': database_name, 'uri': uri}
 
-        # add 'annotation' when RTTM file is provided
-        # defaults to empty Annotation because of
-        # github.com/pyannote/pyannote-database/pull/13#discussion_r261564520)
-        if file_rttm is not None:
-            current_file['annotation'] = annotations.get(
-                uri, Annotation(uri=uri))
-
         # add 'annotated' when UEM file is provided
-        # defaults to empty Timeline for the same reason as above
+        # defaults to empty Timeline because of
+        # github.com/pyannote/pyannote-database/pull/13#discussion_r261564520)
         if file_uem is not None:
-            current_file['annotated'] = annotated.get(uri, Timeline(uri=uri))
+            current_file['annotated'] = annotateds.get(uri, Timeline(uri=uri))
 
+        # add 'annotation' when RTTM file is provided
+        # defaults to empty Annotation for the same reason as above
+        if file_rttm is not None:
+            annotation = annotations.get(uri, Annotation(uri=uri))
+            annotated = current_file.get('annotated')
+            if annotated:
+                if not annotated.covers(annotation.get_timeline()):
```

Comment:
```suggestion
            if annotated and not annotated.covers(annotation.get_timeline()):
```

Reply:
wont `annotated` be `None` if `file_uem is None` ?

Cf. your previous comment 
> This will fail with a `KeyError` if there is no `annotated` section in `database.yml`.